iree/tools/utils/image_util.c

```diff
@@ -0,0 +1,192 @@
+                                           iree_make_cstring_view(" ,"), 0);
+  if (idx <= 0) {
+    return iree_make_status(IREE_STATUS_INVALID_ARGUMENT, "parse input range");
+  }
+  float input_range[2];
+  if (!iree_string_view_atof(iree_string_view_substr(input_range_str, 0, idx),
+                             &input_range[0]) ||
+      !iree_string_view_atof(
+          iree_string_view_substr(input_range_str, idx + 1, INTPTR_MAX),
+          &input_range[1])) {
+    return iree_make_status(IREE_STATUS_INVALID_ARGUMENT, "parse input range");
+  }
+  input_scale[0] = fabsf(input_range[1] - input_range[0]) / 2.0f;
+  input_scale[1] = (input_range[0] + input_range[1]) / 2.0f;
+  return iree_ok_status();
+}
+
+iree_status_t iree_tools_utils_pixel_rescaled_to_buffer(
+    const uint8_t* pixel_data, iree_host_size_t buffer_length, float* buffer) {
+  float input_scale[2] = {0.0f};
+  IREE_RETURN_IF_ERROR(iree_tools_utils_parse_pixel_range(input_scale));
+  const float kUint8Mean = 127.5f;
+  for (int i = 0; i < buffer_length; ++i) {
+    buffer[i] =
+        (((float)(pixel_data[i])) - kUint8Mean) / kUint8Mean * input_scale[0] +
+        input_scale[1];
+  }
+  return iree_ok_status();
+}
+
+iree_status_t iree_tools_utils_load_pixel_data(
+    const iree_hal_dim_t* shape, iree_host_size_t shape_rank,
+    iree_hal_element_type_t element_type, const char* filename,
+    uint8_t** pixel_data, iree_host_size_t* buffer_length) {
+  int img_dims[3];
+  if (stbi_info(filename, img_dims, &(img_dims[1]), &(img_dims[2])) == 0) {
+    return iree_make_status(IREE_STATUS_NOT_FOUND, "can't load image %s",
+                            filename);
+  }
+  if (!(element_type == IREE_HAL_ELEMENT_TYPE_FLOAT_32 ||
+        element_type == IREE_HAL_ELEMENT_TYPE_SINT_8 ||
+        element_type == IREE_HAL_ELEMENT_TYPE_UINT_8)) {
+    return iree_make_status(IREE_STATUS_UNIMPLEMENTED,
+                            "element type not supported");
+  }
+  switch (shape_rank) {
+    case 2: {  // Assume tensor <heightxwidth>
+      if (img_dims[2] != 1 || (shape[0] != img_dims[1]) ||
+          (shape[1] != img_dims[0])) {
+        return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                                "image size: %dx%dx%d, expected: %dx%d",
+                                img_dims[0], img_dims[1], img_dims[2], shape[1],
+                                shape[0]);
+      }
+      break;
+    }
+    case 3: {  // Assume tensor <heightxwidthxchannel>
+      if (shape[0] != img_dims[1] || shape[1] != img_dims[0] ||
+          shape[2] != img_dims[2]) {
+        return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                                "image size: %dx%dx%d, expected: %dx%dx%d",
+                                img_dims[0], img_dims[1], img_dims[2], shape[1],
+                                shape[0], shape[2]);
+      }
+      break;
+    }
+    case 4: {  // Assume tensor <batchxheightxwidthxchannel>
```

Comment:
nit: usespacesbetweenwords :P ? (same for the other tensor layout comments)

```suggestion
    case 4: {  // Assume tensor <batch x height x width x channel>
```

Reply:
Done.